Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware_Tests.swift

```diff
@@ -104,6 +104,54 @@ final class MemberEventMiddleware_Tests: XCTestCase {
         )
     }
     
+    func test_memberAddedEvent_linksNewMember_toMemberListQueries() throws {
+        let cid = ChannelId.unique
+        let newMemberId = UserId.unique
+        
+        // Create MemberAddedEventDTO payload
+        let eventPayload: EventPayload = .init(
+            eventType: .memberAdded,
+            cid: cid,
+            user: .dummy(userId: newMemberId),
+            memberContainer: .dummy(userId: newMemberId),
+            createdAt: .unique
+        )
+        
+        // Create event with payload.
+        let event = try MemberAddedEventDTO(from: eventPayload)
+        
+        // Create query
+        let memberListQuery = ChannelMemberListQuery(cid: cid)
+        let channelPayload = dummyPayload(with: cid, numberOfMessages: 0, includeMembership: false)
+        let existingMember = try XCTUnwrap(channelPayload.members.first)
+        
+        // Create channel and MemberListQuery in the database.
+        try database.writeSynchronously { session in
+            try session.saveChannel(payload: channelPayload)
+            try session.saveMember(payload: existingMember, channelId: cid, query: memberListQuery)
+        }
+        
+        // Load the channel
+        var channel: ChatChannel? {
+            database.viewContext.channel(cid: cid)?.asModel()
+        }
+        
+        // Load the MemberListQueryDTO
+        var memberListQueryDTO: ChannelMemberListQueryDTO? {
+            database.viewContext.channelMemberListQuery(queryHash: memberListQuery.queryHash)
+        }
```

Comment:
nit:
```suggestion
        let channel = try XCTUnwrap(
         database.viewContext.channel(cid: cid)?.asModel()
         )
        
        // Load the MemberListQueryDTO
        let memberListQueryDTO = try XCTUnwrap(
         database.viewContext.channelMemberListQuery(queryHash: memberListQuery.queryHash)
        )
```

Reply:
hmm, are we using the channel at all? 🤔

Reply:
`channel` was a copy/paste leftover :D we can't use `let memberListQuery` since we need it to be loaded again when we call it again